src/lua_details.cc

```diff
@@ -219,33 +237,28 @@ void open_safe_library_subset(sol::state& lua)
     globals["require"] = sol::nil;
 }
 
-std::function<void(sol::this_state, sol::variadic_args)>
-make_print_fct(
-    std::function<void(const std::string&, OptionalStepIndex, CommChannel*)> print_fct)
+void print_fct(sol::this_state sol, sol::variadic_args va)
 {
-    return
-        [print_fct = std::move(print_fct)](sol::this_state sol, sol::variadic_args va)
-        {
-            sol::state_view state{ sol };
-            auto tostring{ state["tostring"] };
-
-            try
-            {
-                gul14::SmallVector<std::string, 8> stringified_args;
-                stringified_args.reserve(va.size());
-
-                for (auto v : va)
-                    stringified_args.push_back(tostring(v));
-
-                print_fct(gul14::join(stringified_args, "\t") + "\n",
-                          get_step_idx_from_registry(sol),
-                          get_comm_channel_ptr_from_registry(sol));
-            }
-            catch (const Error& e)
-            {
-                abort_script_with_error(sol, e.what());
-            }
-        };
+    sol::state_view state{ sol };
+    auto tostring{ state["tostring"] };
+
+    try
+    {
+        gul14::SmallVector<std::string, 8> stringified_args;
+        stringified_args.reserve(va.size());
+
+        for (auto v : va)
+            stringified_args.push_back(tostring(v));
+
+        send_message(Message::Type::output, gul14::join(stringified_args, "\t") + "\n",
```

Comment:
... as for "too long" - could you explain what you mean? In the end the script writer decides how much output she wants to produce...?

Reply:
Say that we have a tab that has at least a length of maximal 8 spaces, where the follow up string fragment is positioned according to the previous non space character that occupies spaces from the 8 tab spaces. For example `"This\tis\ta\tfamous\ttext."` will be printed as

`"This    is      a       famous  text."`

That is what I meant with "too long" because it extracts the output with additional spaces. The same string fragments but now done with a comma-'join' would be smaller:

`This, is, a, famous, text."`

Reply:
I must admit I also do not like the tab-output very much. But it is what standard Lua does (or at least close), so I would not like to confuse Lua developers by sprinkling commas into the output of their `print()` statements.

Reply:
Ok, than we leave it as it is.